### ISSUE TITLE
:c:func:`delay` does not produce a reference

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1285]
I have a C function in my project that is named "delay". For this function, no reference to its documentation is generated when I say :c:func:`delay`. The feature works fine for all other functions, BTW.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1285
- Originally reported by: Ben Franksen
- Originally created at: 2013-10-07T00:58:56.191

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-10-07 03:25:46+00:00_

Can you show us how the definition and the reference look in markup?

**Comment by shimizukawa** [comment]
_From Ben Franksen on 2013-10-07 11:25:47+00:00_

Sure, see below. BTW, I just found that this happens only if the section name above the function definition is the same as the function name. If I change the section name, the problem disappears. However, it does only happen if the name (of the function and section title) is 'delay', otherwise it works.

This is the definition:

```
#!rest

   delay
   ^^^^^

   .. c:function::
      boolean delay(double delay_in_seconds)

```

And this is one instance of a reference:

```
#!rest

   action code, but the effect is probably not what you want. For instance,
   calling :c:func:`delay` in action code does *not* introduce a delay at this
   point. Later versions of *snc* may output a warning or even stop with an
   error.

```

The whole project including documentation sources is in a darcs repo at [http://www-csr.bessy.de/control/SoftDist/sequencer/repo/branch-2-2]()

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-17 06:11:04+00:00_

Closes #1285: Avoid name clashes between C domain objects and section titles.

→ <<cset de92e3b0265e6613278a5dbf83bf833d176f7c03>>
